conn/physic/units_test.go

```diff
@@ -536,3 +538,255 @@ func BenchmarkCelsiusFloatg(b *testing.B) {
+		{"exponet too small for int64", decimal{"123", -20, positive}, 0},
+		{"max*10^1", decimal{"9223372036854775807", 1, positive}, 9223372036854775807},
+		{"-max*10^1", decimal{"9223372036854775807", 1, negative}, -9223372036854775807},
+	}
+
+	for _, tt := range succeeds {
+		got, err := dtoi(tt.in, 0)
+
+		if got != tt.expected {
+			t.Errorf("case dtoi() %s got %v expected %v", tt.name, got, tt.expected)
+		}
+		if err != nil {
+			t.Errorf("case dtoi() %s got an unexpected error %v", tt.name, err)
+		}
+	}
+
+	for _, tt := range fails {
+		got, err := dtoi(tt.in, 0)
+
+		if got != tt.expected {
+			t.Errorf("case dtoi() %s got %v expected %v", tt.name, got, tt.expected)
+		}
+		if err == nil {
+			t.Errorf("case dtoi() %s expected %v but got nil", tt.name, err)
+		}
+	}
+}
+
+func TestPrefix(t *testing.T) {
+	tests := []struct {
+		name   string
+		prefix rune
+		want   prefix
+		n      int
+	}{
+		{"pico", 'p', pico, 1},
+		{"nano", 'n', nano, 1},
+		{"micro", 'u', micro, 1},
+		{"mu", 'µ', micro, 2},
+		{"milli", 'm', milli, 1},
+		{"none", 0, none, 0},
+		{"kilo", 'k', kilo, 1},
+		{"mega", 'M', mega, 1},
+		{"giga", 'G', giga, 1},
+		{"tera", 'T', tera, 1},
+	}
+	for _, tt := range tests {
+		got, n := parseSIPrefix(tt.prefix)
+
+		if got != tt.want || n != tt.n {
+			t.Errorf("wanted prefix %d, and len %d, but got prefix %d, and len %d", tt.want, tt.n, got, n)
+		}
+
+	}
+}
+
+func TestParseError(t *testing.T) {
+	tests := []struct {
+		name string
+		err  error
+		want string
+	}{
+		{"empty", &parseError{s: "", err: nil}, "parse error"},
+		{"empty", &parseError{s: "", err: errors.New("test")}, "parse error: test: \"" + "\""},
+		{"noUnits", noUnits("someunit"), "parse error: no units provided, need: \"someunit\""},
+	}
+	for _, tt := range tests {
+		got := tt.err.Error()
+
+		if got != tt.want {
+			t.Errorf("wanted err string:\n%s but got:\n%s", tt.want, got)
+		}
+
+	}
+}
+func TestMaxInt64(t *testing.T) {
+	if strconv.FormatUint(maxInt64, 10) != maxUint64Str {
+		t.Fatal("unexpected text representation of max")
+	}
+}
+
+func BenchmarkDecimal(b *testing.B) {
+	var d decimal
+	var n int
+	var err error
+	for i := 0; i < b.N; i++ {
+		if d, n, err = atod("337.2m"); err != nil {
+			b.Fatal(err)
+		}
+	}
+	b.StopTimer()
+	_ = fmt.Sprintf("%v %d", d, n)
+}
+
+func BenchmarkString2Decimal2Int(b *testing.B) {
+
+	var d decimal
+	var n int
+	var err error
+	var v int64
+	for i := 0; i < b.N; i++ {
+		if d, n, err = atod("337.2m"); err != nil {
+			b.Fatal(err)
+		}
+		if v, err = dtoi(d, 0); err != nil {
+			b.Fatal(err)
+		}
+	}
+	b.StopTimer()
+	_ = fmt.Sprintf("%d %d", v, n)
+}
+
+func BenchmarkDecimalNeg(b *testing.B) {
+	var d decimal
+	var n int
+	var err error
+	for i := 0; i < b.N; i++ {
+		if d, n, err = atod("-337.2m"); err != nil {
+			b.Fatal(err)
+		}
+	}
+	b.StopTimer()
+	_ = fmt.Sprintf("%v %d", d, n)
+}
+
+func BenchmarkString2Decimal2IntNeg(b *testing.B) {
+
```

Comment:
Similar here, I'm not used to an empty line after an opening bracket. In practice I mostly favor consistency, so have it consistent across the functions.

Reply:
fwiw: It's simply does not catch my eye as an error, I had to read bottom to top to see any of them.
I'll keep a closer eye on my formatting style, as I can imagine it is a pain to nit them.